cmd/launcher/svc_config_windows.go

```diff
@@ -213,10 +223,30 @@ func setRecoveryActions(ctx context.Context, logger *slog.Logger, service *mgr.S
 		},
 	}
 
+	// If the recovery actions are already set, we don't need to do anything
+	if recoveryActionsAreSet(curRecoveryActions, recoveryActions) {
+		return
+	}
+
 	if err := service.SetRecoveryActions(recoveryActions, 24*60*60); err != nil { // 24 hours
 		logger.Log(ctx, slog.LevelError,
 			"setting RecoveryActions",
 			"err", err,
 		)
 	}
 }
+
+// recoveryActionsAreSet checks if the current recovery actions are set to the desired recovery actions
+func recoveryActionsAreSet(curRecoveryActions []mgr.RecoveryAction, recoveryActions []mgr.RecoveryAction) bool {
+	if curRecoveryActions != nil {
+		if len(curRecoveryActions) == len(recoveryActions) {
+			for i := range curRecoveryActions {
+				if curRecoveryActions[i].Type != recoveryActions[i].Type && curRecoveryActions[i].Delay != recoveryActions[i].Delay {
+					return false
+				}
+			}
+			return true
+		}
+	}
+	return false
```

Comment:
We prefer to reduce nesting when possible for readability -- you can modify this to do some early checks and returns to accomplish this:

```suggestion
        if curRecoveryActions == nil || len(curRecoveryActions != len(recoveryActions) {
                return false
        }
	for i := range curRecoveryActions {
		if curRecoveryActions[i].Type != recoveryActions[i].Type && curRecoveryActions[i].Delay != recoveryActions[i].Delay {
			return false
		}
	}
	return true
```

Reply:
That is much easier to read. Thanks for the suggestion!